services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerEndpoint.java

```diff
@@ -606,9 +634,26 @@ private VCIssuanceContext getVCToSign(List<OID4VCMapper> protocolMappers, Suppor
 
         LOGGER.debugf("The credential to sign is: %s", vc);
 
-        return new VCIssuanceContext().setAuthResult(authResult)
-                .setVerifiableCredential(vc)
+        // Build format-specific credential
+        CredentialBody credentialBody = locateCredentialBuilder(credentialConfig)
+                .buildCredentialBody(vc, credentialConfig.getCredentialBuildConfig());
+
+        return new VCIssuanceContext()
+                .setAuthResult(authResult)
+                .setCredentialBody(credentialBody)
                 .setCredentialConfig(credentialConfig)
                 .setCredentialRequest(credentialRequestVO);
     }
+
+    private CredentialBuilder locateCredentialBuilder(SupportedCredentialConfiguration credentialConfig) {
+        String format = credentialConfig.getFormat();
+        String locator = AbstractCredentialBuilder.computeLocator(format);
+        CredentialBuilder credentialBuilder = credentialBuilders.get(locator);
+
+        if (credentialBuilder == null) {
+            throw new IllegalArgumentException(String.format("No credential builder found for format %s", format));
```

Comment:
Since credential builders are looked up in a request to response flow, it is indeed possible to catch the exception in the controller and treat it explicitly in this way. However, I think it is preferable to throw an exception that the web stack can understand and convert into a response implicitly. In this regard, `BadRequestException`'s are already thrown at several places in the class. In response to your concern, I then changed `IllegalArgumentException` to `BadRequestException`. Do you find this update satisfactory?

Reply:
Yes, Thank you. I confirmed it.